internal/datasystem/fdv2_store.go

```diff
@@ -0,0 +1,226 @@
+
+	// Represents the store that all flag/segment data queries are served from after data is received from
+	// initializers/synchronizers. Before the in-memory store is initialized, queries are served from the
+	// persistentStore (if configured).
+	memoryStore subsystems.DataStore
+
+	active subsystems.DataStore
+
+	// Whether the memoryStore's data should be considered authoritative, or fresh - that is, if it is known
+	// to be the latest data. Data from a baked in file for example would not be considered refreshed. The purpose
+	// of this is to know if we should commit data to the persistentStore. For example, if we initialize with "stale"
+	// data from a local file (refreshed=false), we may not want to pollute a connected Redis database with it.
+	// TODO: this could also be called "Authoritative". "It was the latest at some point.. that point being when we asked
+	// if it was the latest".
+	availability DataAvailability
+
+	// Protects the refreshed, persistentStore, persistentStoreMode, and active fields.
+	mu sync.RWMutex
+
+	loggers ldlog.Loggers
+}
+
+// NewStore creates a new store. By default the store is in-memory. To add a persistent store, call SwapToPersistent. Ensure this is
+// called at configuration time, only once and before the store is ever accessed.
+func NewStore(loggers ldlog.Loggers) *Store {
+	s := &Store{
+		persistentStore:     nil,
+		persistentStoreMode: subsystems.StoreModeRead,
+		memoryStore:         datastore.NewInMemoryDataStore(loggers),
+		availability:        Defaults,
+		loggers:             loggers,
+	}
+	s.active = s.memoryStore
+	return s
+}
+
+// Close closes the store. If there is a persistent store configured, it will be closed.
+func (s *Store) Close() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.persistentStore != nil {
+		return s.persistentStore.Close()
+	}
+	return nil
+}
+
+// GetActive returns the active store, either persistent or in-memory. If there is no persistent store configured,
+// the in-memory store is always active.
+func (s *Store) getActive() subsystems.DataStore {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.active
+}
+
+// DataAvailability returns the status of the store's data. Defaults means there is no data, Cached means there is
+// data, but it's not guaranteed to be recent, and Refreshed means the data has been refreshed from the server.
+func (s *Store) DataAvailability() DataAvailability {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.availability
+}
+
+// Mirroring returns true data is being mirrored to a persistent store.
+func (s *Store) mirroring() bool {
+	return s.persistentStore != nil && s.persistentStoreMode == subsystems.StoreModeReadWrite
+}
+
+// nolint:revive // Standard DataSourceUpdateSink method
+func (s *Store) Init(allData []ldstoretypes.Collection, dataStatus datastatus.DataStatus) bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// TXNS-PS: Requirement 1.3.3, must apply updates to in-memory before the persistent Store.
```

Comment:
Did we settle on a format for this? Maybe it would be easier if we went with `TXNS-PS(1.3.3): whatever`.